### PR TITLE
[VL] Cleanup substrait to velox plan

### PR DESCRIPTION
**Pull request by ulysses-you**
## What changes were proposed in this pull request?

- Remove the duplicate branch in `toVeloxPlan(const ::substrait::Rel& rel)`
- Use the `processSortField` to transform the window sort order

## How was this patch tested?

PASS CI

### CODE REVIEW

File: cpp/core/shuffle/ShuffleWriter.cc [hunk 1]
```diff
@@ -57,7 +57,7 @@ class ShuffleBufferPool::MemoryPoolWrapper : public arrow::MemoryPool {
     return status;
   }
 
-  void Free(uint8_t* buffer, int64_t size, int64_t alignment) {
+  void Free(uint8_t* buffer, int64_t size, int64_t alignment) override {
```
Comment (ulysses-you) [inline cpp/core/shuffle/ShuffleWriter.cc:60]: I do not want to touch irrelevant code, but it breaks the compilation on macos..

**Comment by ulysses-you** [comment]
cc @marin-ma @PHILO-HE thank you

**Review by yaooqinn — APPROVED** [review]

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks!

**Comment by ulysses-you** [comment]
thank you for review, merging to main

**Pull request merged**